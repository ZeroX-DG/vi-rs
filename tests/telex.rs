mod shared;

fn snapshot_transform(lines: &str) -> String {
    shared::transform_lines(lines, |word| {
        let mut trasformed_word = String::new();
        vi::transform_buffer(&vi::TELEX, word.chars(), &mut trasformed_word);
        trasformed_word
    })
}

macro_rules! gen_test_telex {
    ($name:tt, $path: tt) => {
        gen_test!(snapshot_transform, $name, $path);
    };
}
gen_test_telex!(simple_telex, "../testdata/input/simple_telex.txt");
gen_test_telex!(
    non_vietnamese_telex,
    "../testdata/input/non_vietnamese_telex.txt"
);
