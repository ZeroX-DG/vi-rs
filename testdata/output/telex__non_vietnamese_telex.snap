---
source: tests/telex.rs
expression: snapshot_transform(contents)
info:
  input_file: "../testdata/input/non_vietnamese_telex.txt"
---
samuwrite
cáe
case
cat
doggo
land
overflow
ơd
old
dd
ddd
dddddddddd
ww
